src/vmc/webhook/thehive/conventer.py

```diff
@@ -0,0 +1,146 @@
+        self._hive_client = hive_client
+
+    def process(self, task):
+        alert = self._hive_client.get_alert(task.alert_id)
+        for art in alert['artifacts']:
+            if 'url' in art['dataType']:
+                task.scan_url = art['data'].replace(' ', '')
+            if 'business-unit' in art['dataType']:
+                task.tenant = art['data']
+            if 'ip' in art['dataType']:
+                task.ip = art['data']
+            if 'document-id' in art['dataType']:
+                task.document_id = art['data']
+
+        task.title, task.group = self.get_task_title_and_group(alert['title'])
+        task.source = alert['source']
+        task.description = alert['description']
+        task.save()
+
+    @staticmethod
+    def get_task_title_and_group(title):
+        new_title = ''.join(title.split(':')[1:])
+
+        if 'critical' in title:
+            return new_title, 'critical'
+        if 'high' in title:
+            return new_title, 'high'
+        if 'medium' in title:
+            return new_title, 'medium'
+
+        return new_title, 'low'
+
+
+class CaseManager:
+
+    def __init__(self, hive_client):
+        self._hive = hive_client
+        self._updated = set()
+
+    def get_or_create_case(self, scan_url, source, tenant):
+        case = Case.objects.filter(scan_url=scan_url).first()
+        if not case:
+
+            if tenant:
+                title = F'New scan from {source} for {tenant}'
+            else:
+                title = F'New scan from {source}'
+
+            case_id = self._hive.create_case(title, title)
+            case = Case.objects.create(id=case_id, scan_url=scan_url, tenant=tenant)
+
+            return case
+        return case
+
+    def merge_alert_to_case(self, case, task):
+        try:
+            LOGGER.debug(F'Merging case {case.id} to task {task.alert_id}')
+            self._hive.merge_alert_to_case(task.alert_id, case.id)
+            task.case = case
+            task.task_id = self._hive.create_task(case.id, task.title, task.description, task.group)
+            task.save()
+            self._updated.add(case.id)
+        except Exception as ex:
+            LOGGER.error(ex)
+
+    def update_cases_desc(self):
+        for case_id in self._updated:
+            obj = Case.objects.get(id=case_id)
+            tags = {obj.tenant}
+            tags.update({t.ip for t in obj.task_set.all()})
+
+            self._hive.update_case(case_id,
+                description=self._build_description(obj),
+                tags=tags
+            )
+        self._updated = {}
+
+    @staticmethod
+    def _build_description(obj):
+        groups = {}
+        for task in obj.task_set.all():
+            if task.group not in groups:
+                groups.update({task.group: 1})
+            else:
+                groups[task.group] += 1
+
+        s = F"""**Summary**:\n
+Critical: {groups['critical'] if 'critical' in groups else 0}\n
```

Comment:
From this point we should consider moving that to separate file (with ssome constant values or strings) and concatenate